docs/model-dev-guide/api-guides/apis-howto/api-keras-ug.rst

```diff
@@ -94,3 +94,41 @@ implement the :class:`determined.keras.callbacks.Callback` interface (an extensi
 ``tf.keras.callbacks.Callbacks`` interface) and supply them to the
 :class:`~determined.keras.TFKerasTrial` by implementing
 :meth:`~determined.keras.TFKerasTrial.keras_callbacks`.
+
+***********
+ Profiling
+***********
+
+Determined supports integration with the native TF Keras profiler. Results will automatically be
+uploaded to Tensorboard, which is accessible from trial page on the Determined UI.
+
+The Keras profiler is configured as a callback in the :class:`~determined.keras.TFKerasTrial` class.
+The :class:`determined.keras.callbacks.TensorBoard` callback is a thin wrapper around the native
+Keras Tensorboard callback, ``tf.keras.callbacks.TensorBoard``. It overrides the ``log_dir``
+argument to the Determined Tenosrboard path, while other arguments are passed directly into
+``tf.keras.callbacks.TensorBoard``. See the `official Keras API documentation
+<https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard>`_ for a complete list of
+accepted arguments.
+
+The following code snippet will configure profiling for batches 5 and 10, and will compute weight
+histograms every 1 epochs.
+
+.. code:: python
+
+   from determined import keras
+
+   def keras_callbacks(self) -> List[tf.keras.callbacks.Callback]:
+      return [
+          keras.callbacks.TensorBoard(
+              update_freq="batch",
+              profile_batch='5, 10',
+              histogram_freq=1,
+          )
+      ]
+
+.. note::
+
+   Though specifying batches to profile with ``profile_batch`` is optional, profiling every batch
+   may cause a large amount of data to be uploaded to Tensorboard. This may result in long rendering
+   times for Tensorboard and memory issues. For long-running experiments, it is recommended to
+   configure profiling only on desired batches.
```

Comment:
***********
 Profiling
***********

Determined supports integration with the native TF Keras profiler. Results are automatically uploaded to TensorBoard, accessible from the trial page in the WebUI.

Configuring the Keras Profiler
==============================

The Keras profiler is configured as a callback in the :class:`~determined.keras.TFKerasTrial` class. The :class:`determined.keras.callbacks.TensorBoard` callback is a thin wrapper around the native Keras TensorBoard callback, :class:`tf.keras.callbacks.TensorBoard`. It overrides the ``log_dir`` argument to set the Determined TensorBoard path, while other arguments are passed directly into :class:`tf.keras.callbacks.TensorBoard`. For a list of accepted arguments, consult the `official Keras API documentation
<https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard>`_.

Example Code Snippet
====================

This code snippet shows you how to configure profiling for 
batches 5 and 10, computing weight histograms every 1 epochs.

.. code-block:: python

   from determined import keras
   from typing import List
   import tensorflow as tf

   def keras_callbacks(self) -> List[tf.keras.callbacks.Callback]:
       return [
           keras.callbacks.TensorBoard(
               update_freq="batch",
               profile_batch='5, 10',
               histogram_freq=1,
           )
       ]

.. note::

   While specifying batches to profile with ``profile_batch`` is optional, profiling every batch can generate a large amount of data, causing long rendering times and potential memory issues in TensorBoard. For long-running experiments, it's advised to profile only specific batches.